tests/Credentials/CredentialProviderTest.php

```diff
@@ -277,6 +277,80 @@ public function testEnsuresFileIsNotEmpty()
         }
     }
 
+    public function testCreatesFromProcessCredentialProvider()
+    {
+        $dir = $this->clearEnv();
+        $ini = "[foo]\ncredential_process = "
+            . "echo '{\"AccessKeyId\":\"foo\",\"SecretAccessKey\":\"bar\", \"Version\":1}'";
+        file_put_contents($dir . '/credentials', $ini);
+        putenv('HOME=' . dirname($dir));
+
+        $creds = call_user_func(CredentialProvider::process('foo'))->wait();
+        $this->assertEquals('foo', $creds->getAccessKeyId());
+        $this->assertEquals('bar', $creds->getSecretKey());
+    }
+
+    /**
+     * @expectedException \Aws\Exception\CredentialsException
+     * @expectedExceptionMessage No credential_process present in INI profile
+     */
+    public function testEnsuresProcessCredentialIsPresent()
+    {
+        $dir = $this->clearEnv();
+        $ini = "[default]\naws_access_key_id = foo\n"
+            . "aws_secret_access_key = baz\n[foo]";
+        file_put_contents($dir . '/credentials', $ini);
+        putenv('HOME=' . dirname($dir));
+
+        try {
+            $creds = call_user_func(CredentialProvider::process())->wait();
+        } catch (\Exception $e) {
+            unlink($dir . '/credentials');
```

Comment:
Should the `unlink` be moved to a [`finally` block](https://www.php.net/manual/en/language.exceptions.php#language.exceptions.finally) in case the test fails by not getting to the exception? Applies to the following tests as well.

Reply:
Yes. Moved to a ```finally``` block.